src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,420 @@
+    try:
+        docker_command = get_docker_command()
+    except CLIError:
+        accumulated_errors |= _error_handler(Errors.DOCKER_COMMAND_ERROR,
+                                             "Obtain docker command",
+                                             ignore_errors,
+                                             "DOCKER OPERATIONS FAILED")
+        return accumulated_errors
+
+    output = getoutput(docker_command + ' system info')
+
+    # Docker daemon check
+    if output.find(DOCKER_DAEMON_NOT_RUNNING) != -1:
+        accumulated_errors |= _error_handler(Errors.DOCKER_DAEMON_ERROR, "Docker daemon status", ignore_errors)
+    else:
+        daemon_status = 'available'
+        print('Docker daemon status: {}'.format(daemon_status))
+
+    # Docker version check
+    lines = output.splitlines()
+    regex = re.compile(r'([0-9.b]+)')
+    docker_version = ""
+
+    for line in lines:
+        if line.startswith("Server Version"):
+            try:
+                docker_version = regex.search(line).group(1)
+            except AttributeError:
+                pass
+
+    if docker_version in ("", None):
+        accumulated_errors |= _error_handler(Errors.DOCKER_VERSION_ERROR, "Docker server version", ignore_errors)
+    else:
+        print('Docker server version: {}'.format(docker_version))
+
+    return accumulated_errors
+
+def _get_cli_version(ignore_errors):
+    accumulated_errors = 0
+
+    #ACR CLI check
+    output = getoutput('az --version')
+
+    lines = output.splitlines()
+    acr_cli_version = ""
+    regex = re.compile(r'([0-9.b]+)')
+
+    for line in lines:
+        if line.startswith('acr'):
+            try:
+                acr_cli_version = regex.search(line).group(1)
+            except AttributeError:
+                pass
+
+    if acr_cli_version in ("", None):
+        accumulated_errors |= _error_handler(Errors.ACR_CLI_VERSION_ERROR, "ACR CLI version", ignore_errors)
+    else:
+        print('ACR cli version: {}'.format(acr_cli_version))
+
+    return accumulated_errors
+
+def _get_helm_version(ignore_errors):
+    accumulated_errors = 0
+
+    # If there's not helm command there's no reason to proceed with the operations
+    try:
+        helm_command = _get_helm_command()
+    except CLIError:
+        accumulated_errors |= _error_handler(Errors.HELM_COMMAND_ERROR,
+                                             "Obtain helm command",
+                                             ignore_errors,
+                                             "HELM OPERATIONS FAILED")
+        return accumulated_errors
+
+    # Helm check
+    output = getoutput(helm_command + ' version')
+
+    lines = output.splitlines()
+    regex = re.compile(r'SemVer:"([0-9.vb]+)"')
+    client_version = ""
+    server_version = ""
+
+    for line in lines:
+        if line.startswith('Client'):
+            try:
+                client_version = regex.search(line).group(1)
```

Comment:
Yes, this is a future planning. Also, doesn't this go against the idea of making the output consistent?
I agree that this can get obsoleted if there are changes on the client display, but maybe we can incorporate a lot of regex to cover multiple versions

Reply:
I am not fully sure with such context parsing. If you don't plan to do anything with the version number, do we still need extract it and come up with a new message? Can we just print the raw string as is?

Your code can be easily broken if helm (or docker, for that matter) makes a change to print a different string.

Reply:
I see. Makes sense, sounds the best idea for now.

Reply:
Adding this in next push.